### ISSUE TITLE
Add the ability to change accent color

### ISSUE THREAD

**Issue by DanyGee** [keepassxreboot/keepassxc#5335]
## Summary
[TIP]:  # ( DO NOT include screenshots of your actual database! )
[NOTE]: # ( Provide a brief overview of what the new feature is all about )
Allow selection of an accent color, so users can change the default green to color of their choice (either a full spectrum color chooser or some preinstalled few ones should do).

## Examples
[NOTE]: # ( Show us a picture or mock-up of your proposal )
![keepassxc_accent_blue](https://user-images.githubusercontent.com/26818551/91103082-e6b89600-e66a-11ea-942f-1b4fa236982f.png)

![keepassxc_accent_violet](https://user-images.githubusercontent.com/26818551/91103087-e9b38680-e66a-11ea-8d57-4f81b2e2088f.png)

Color picker might look like this:
![color_picker1](https://user-images.githubusercontent.com/26818551/91103273-67779200-e66b-11ea-9163-63198d22e5ab.png)

Eventually like this:
![color_picker2](https://user-images.githubusercontent.com/26818551/91103283-6e060980-e66b-11ea-972f-1e2f468d3899.png)

## Context
[NOTE]: # ( Why does this feature matter to you? What unique circumstances do you have? )
The context is purely aesthetic.
User might want to match the apps accent with his desktop theme or match the colors of his company he owns or works in.
As minor as it may seem, it will still be value added 💡

**Comment by droidmonkey** [comment]
Not a bad idea

**Comment by pkrc267** [comment]
I would like to work on it. Have basic understanding of Qt framework. Will be a good learning experience.

**Comment by droidmonkey** [comment]
Good luck!

**Comment by pkrc267** [comment]
Need a little help. I have gone through CONTRIBUTING.md,  forked and cloned the repository. How can I open the project in Qt creator?

**Comment by droidmonkey** [comment]
Open the cmakelists.txt

**Comment by pkrc267** [comment]
So I have been a bit occupied with my college curriculum. But I'm back and here is a little progress that I've made.
I'm aware of a few points like position of the Color button, but that I think can be changed with some help. For now how does this look. Also please tell me what can/should be done.

_p.s. I added the gradient as it looked cool. But have to work on text color change accordingly._

**Clear video:** [https://drive.google.com/file/d/1IdX9SJIJo0vZCIEhYLYHpYpJeqPhXbYC/view?usp=sharing](url)

**GIF:**
![ezgif com-video-to-gif](https://user-images.githubusercontent.com/8735394/95678450-4c94b900-0bea-11eb-8c9d-fc31b299d27b.gif)

**Screenshot:**
<img src="https://user-images.githubusercontent.com/8735394/95678436-34249e80-0bea-11eb-81d1-cc7711b51560.jpg" width="500">

**Comment by droidmonkey** [comment]
Its an experiment, but that would never be merged in unfortunately. Keep learning, you're making progress.

**Comment by watchpocket** [comment]
Adding this feature isn't just an esthetic thing.  

In a list of 40 or 50 entries, all of which may have different background colors, that green highlight color does not stand out enough to show you, at a glance, where you are.  

It doesn't show you well enough which entry you're on (unless you look really hard and click around on a few other entries to see where that green moves to).

(I can't tell if what pkrc267 is working on is what I'm talking about. I just need to create an entry-background highlight color that allows me to distinguish it from all the other entry-background colors. If I could make that green a bright red, for example, it would be a BIG user-interface help.  (since I don't use bright red for the background of any other entry.)

In almost every other respect, KeePassXC is a fantastic app. 

So when you say, @droidmonkey, that this feature will "never be merged," are you saying that the ability to change the entry-highlight color is never going to be added to program?

**Comment by AlexDeFoc** [comment]
I have an idea. I woukd be a good start to have the accent color of the apps listed somehow in a file. And you can edit it. (if it doesnt have the ability to inject other malware or other things thorund the file, when the app reading the files, huh)

**Comment by MathieuDR** [comment]
I think theming in general could use an overhaul. Eg setting up your own colour schemes to use something like catppuccin would be amazing :)

**Comment by yadav-prakhar** [comment]
I had stopped working on this long back due to my limited sense of design skills and this being experimental feature. But it was fun though.
If anyone would like to take up and work on a revamp, it would be nice. If it is aesthetic enough to attract attention it might get merged. You never know.

**Comment by arigit** [comment]
> I think theming in general could use an overhaul. Eg setting up your own colour schemes to use something like catppuccin would be amazing :)

Came here to ask for support for custom themes. The "Native" integration on modern gnome desktops is based on GTK2, it uses the main desktop colors but the window borders look really bad, e.g. I use windows controls in the left side and the keepassxc window has them on the right (when using "Native integration"), a bit of an eyesore. I like the packaged theme, it worked well with default gnome/ubuntu dark, but since I switched my gnome shell theme to Nord, the theme looks really bad, so I had to move to "Native". 

Anyways, theming pains could be sorted out with the support of some customization of the integrated theme, can this please be added.

**Comment by droidmonkey** [comment]
Window controls are not managed by us, they are done by your compositor (desktop env)

**Comment by arigit** [comment]
@droidmonkey  yes, and somehow in the embedded theme, the window controls position follows the desktop setting (left or right), however the theme colors do not match the desktop.

On the other hand, with "native" theming KDE apps on GNOME follow the GTK2 theme, meaning that the colors of the window follow the desktop setting, but the window controls look awful and in the wrong place.

**Comment by droidmonkey** [comment]
Native theme uses the built-in Qt theming plugins that may be triggering a different configuration in KDE. Just a shot in the dark, but Linux theming is God awful.

**Comment by arigit** [comment]
> > I think theming in general could use an overhaul. Eg setting up your own colour schemes to use something like catppuccin would be amazing :)
> 
> Came here to ask for support for custom themes. The "Native" integration on modern gnome desktops is based on GTK2, it uses the main desktop colors but the window borders look really bad, e.g. I use windows controls in the left side and the keepassxc window has them on the right (when using "Native integration"), a bit of an eyesore. I like the packaged theme, it worked well with default gnome/ubuntu dark, but since I switched my gnome shell theme to Nord, the theme looks really bad, so I had to move to "Native".
> 
> Anyways, theming pains could be sorted out with the support of some customization of the integrated theme, can this please be added.

Reply-to-self: 

`env QT_QPA_PLATFORM=xcb keepassxc`

Prior to this, install the package qt5ct, run qt5ct and set it to "gtk2". Probably , you will need to log out and log back in. With the local deb install of keepassxc, this gives what I was trying to get:

![Image](https://github.com/user-attachments/assets/0763c61f-eb7f-46c7-806b-956c49ece449)

With the flatpak, the window controls and borders are correct, but the app colors, fonts etc are KDE's default == eyesore

Lastly - fix the Exec lines in the .desktop files in:
/usr/share/applications/org.keepassxc.KeePassXC.desktop
/home/your_user/.config/autostart/org.keepassxc.KeePassXC.desktop
